### PR TITLE
Gateway http e2e

### PR DESCRIPTION
**Pull request by beingStrongeryqqq**
**What this PR does / why we need it**:
gateway-lwm2m http e2e test
**Will this PR make the community happier**? 
yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [ ] unit test
- [x] e2e test
- [ ] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: azure-pipelines/azure-pipelines.yml [hunk 1]
```diff
@@ -573,6 +653,24 @@ stages:
...
+              command: login
+              containerRegistry: dockerhub-connection
+          - script: |
+              docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+              docker buildx create --use
+            displayName: configure multi-arch and buildx
+          - script: |
+              make buildx-push-image-deviceshifu-gateway-lwm2m
+            displayName: build deviceshifu-gateway-lwm2m
+
```
Comment (rhoninl) [inline azure-pipelines/azure-pipelines.yml:673]: why this pr include push gateway?

File: examples/lwm2m_gw_http/lw/lwm2m-deviceshifu-deployment.yaml
Comment (rhoninl) [inline examples/lwm2m_gw_http/lw/lwm2m-deviceshifu-deployment.yaml:1]: rename this folder, lw is hard to understand

**Review by tomqin93 — CHANGES_REQUESTED** [review]
@beingStrongeryqqq please sync this branch with main

File: azure-pipelines/azure-pipelines.yml [hunk 2]
```diff
@@ -300,6 +300,86 @@ stages:
               kubectl delete -f $(System.DefaultWorkingDirectory)/examples/deviceshifu/demo_device/edgedevice-socket/mock-device
             displayName: "Shifu demo device Socket E2E test"
 
+      - job: kind_e2e_test_lwm2m_with_security
```
Comment (tomqin93) [inline azure-pipelines/azure-pipelines.yml:303]: need to sync with main branch

File: azure-pipelines/azure-pipelines.yml [hunk 3]
```diff
@@ -300,6 +300,86 @@ stages:
...
+              version: $(goVersion)
+          - script: |
+              tag=`cat version.txt` && echo "##vso[task.setvariable variable=tag]$tag"
+            displayName: Set the tag name as an environment variable
+          - script: |
+              make buildx-build-image-shifu-controller
+            displayName: build edgehub/shifu-controller
+          - script: |
+              make buildx-build-image-deviceshifu-http-http
+            displayName: build edgehub/deviceshifu-http-lwm2m
```
Comment (tomqin93) [inline azure-pipelines/azure-pipelines.yml:355]: http-http

File: examples/deviceshifu/mockdevice/lwm2m-http/checkoutput.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Exit immediately if a command exits with a non-zero status
+set -e
+
+# Get the pod name of deviceshifu
+pod_name=$(kubectl get pods -n deviceshifu -l app=deviceshifu-thermometer-deployment -o jsonpath='{.items[0].metadata.name}')
+
+# Execute a curl command inside the nginx pod in the deviceshifu namespace to read the value from the thermometer device service
```
Comment (tomqin93) [inline examples/deviceshifu/mockdevice/lwm2m-http/checkoutput.sh:9]: devices namespace

File: examples/deviceshifu/mockdevice/lwm2m-http/checkoutput.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+out=$(kubectl exec -n deviceshifu nginx -- curl --retry 5 --retry-delay 3 --max-time 15 --connect-timeout 5 thermometer.devices.svc.cluster.local:11111/read_value)
```
Comment (tomqin93) [inline examples/deviceshifu/mockdevice/lwm2m-http/checkoutput.sh:10]: need to read from `deviceshifu-lwm2m`, not from the mockdevice

**Review by tomqin93 — CHANGES_REQUESTED** [review]
@beingStrongeryqqq added some comments

File: azure-pipelines/azure-pipelines.yml (hunk 2 above)
Comment (beingStrongeryqqq) [reply azure-pipelines/azure-pipelines.yml:303]: after sync, it will be retained. I deleted it directly.

**Review by tomqin93 — CHANGES_REQUESTED** [review]
@beingStrongeryqqq added 2 comments, overall looks good

**Review by tomqin93 — APPROVED** [review]

**Review by rhoninl — APPROVED** [review]
LGTM

**Pull request merged**